services/web/client/source/class/osparc/Application.js

```diff
@@ -104,10 +104,13 @@ qx.Class.define("osparc.Application", {
         if (urlFragment.nav[0] === "study" && urlFragment.nav.length > 1) {
           // Route: /#/study/{id}
           osparc.utils.Utils.cookie.deleteCookie("user");
-          osparc.auth.Manager.getInstance().validateToken(() => {
-            osparc.store.Store.getInstance().setCurrentStudyId(urlFragment.nav[1]);
-            this.__loadMainPage();
-          }, this.__loadLoginPage, this);
+          osparc.auth.Manager.getInstance().validateToken()
```

Comment:
Q? So, ``validateToken()`` returns some kind of future (Promise?) that has ``then`` and ``catch`` members where you can hook functions depending the request response?

The code is far more readable!

Very nice!

Reply:
Correct, the `then` and `catch` of a Promise are executed after `resolve` and `reject`, respectively, within the promise. There is also a `finally` that will be called no matter what happened in the Promise.